domain/timeline/src/commonMain/kotlin/social/androiddev/domain/timeline/HomeTimelineRepository.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * This file is part of Dodo.
+ *
+ * Dodo is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ *
+ * Dodo is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Dodo. If not, see <https://www.gnu.org/licenses/>.
+ */
+package social.androiddev.domain.timeline
+
+import kotlinx.coroutines.flow.Flow
+import org.mobilenativefoundation.store.store5.StoreResponse
+import social.androiddev.domain.timeline.model.StatusLocal
+
+interface HomeTimelineRepository {
+    fun read(
+        feedType: FeedType,
+        refresh: Boolean = false
+    ): Flow<StoreResponse<List<StatusLocal>>>
+}
+
+sealed class FeedType(val type: String) {
+    object Home : FeedType("HOME")
```

Comment:
I think the domain model FeedType could just be an Enum. And you can map this Enum -> JSON string that needs to be passed to the API in the RealHomeTimelineRepostory.

Just a thought, I don't feel strongly.

Reply:
let me do as follow up, I had trouble mapping the enum mostly from not working with enums for too long :-)